support-frontend/assets/helpers/abTests/__tests__/abtestTest.ts

```diff
@@ -21,840 +18,303 @@ jest.mock('ophan', () => ({
-					GB: {
-						offset: 0,
-						size: 1,
-					},
-				},
-				isActive: true,
-				referrerControlled: true,
-				seed: 0,
-			},
+			t: buildTest({ variants: [buildVariant({ id: 'control' })] }),
 		};
-		const country = 'GB';
+
 		const participations: Participations = abInit(
 			country,
-			GBPCountries,
-			emptySettings as Settings,
+			countryGroupId,
+			emptySettings,
 			tests,
+			mvt,
 		);
+
 		const expectedParticipations: Participations = {
-			mockTest: 'control',
+			t: 'control',
 		};
+
 		expect(participations).toEqual(expectedParticipations);
 	});
 
-	it('The referrerControlled user should be allocated to the variant bucket', () => {
-		document.cookie = 'GU_mvt_id=12345';
-		window.history.pushState({}, 'Test Title', mockTestVariant);
+	it('uses the variant assignment in the acquisitionData for referrerControlled tests', () => {
 		const tests = {
-			mockTest: {
-				type: 'OTHER',
+			t1: buildTest({
 				variants: [
-					{
-						id: 'control',
-					},
-					{
-						id: 'variant',
-					},
+					buildVariant({ id: 'control' }),
+					buildVariant({ id: 'variant' }),
 				],
-				audiences: {
-					GB: {
-						offset: 0,
-						size: 1,
-					},
-				},
-				isActive: true,
 				referrerControlled: true,
-				seed: 0,
-			},
-		};
-		const country = 'GB';
-		const participations: Participations = abInit(
-			country,
-			GBPCountries,
-			emptySettings as Settings,
-			tests,
-		);
-		const expectedParticipations: Participations = {
-			mockTest: 'variant',
-		};
-		expect(participations).toEqual(expectedParticipations);
-	});
-
-	it('The user should be allocated to the variant bucket', () => {
-		document.cookie = 'GU_mvt_id=12346';
-		const tests = {
-			mockTest: {
-				type: 'OTHER',
+			}),
+			t2: buildTest({
 				variants: [
-					{
-						id: 'control',
-					},
-					{
-						id: 'variant',
-					},
+					buildVariant({ id: 'control' }),
+					buildVariant({ id: 'variant' }),
 				],
-				audiences: {
-					GB: {
-						offset: 0,
-						size: 1,
-					},
-				},
-				isActive: true,
-				referrerControlled: false,
-				seed: 6,
-			},
+				referrerControlled: true,
+			}),
 		};
-		const country = 'GB';
-		const countryGroupId = GBPCountries;
+
+		const acquisitionAbTests = [
+			buildAcquisitionAbTest({ name: 't1', variant: 'control' }),
+			buildAcquisitionAbTest({ name: 't2', variant: 'variant' }),
+		];
+
 		const participations: Participations = abInit(
 			country,
 			countryGroupId,
-			emptySettings as Settings,
+			emptySettings,
 			tests,
+			mvt,
+			acquisitionAbTests,
 		);
+
 		const expectedParticipations: Participations = {
-			mockTest: 'variant',
+			t1: 'control',
+			t2: 'variant',
 		};
+
 		expect(participations).toEqual(expectedParticipations);
 	});
 
-	it('The user should be allocated to the control bucket', () => {
-		document.cookie = 'GU_mvt_id=12346';
+	it('uses the variant assignment in the acquisitionData for referrerControlled tests belonging to a campaign', () => {
```

Comment:
This is a new test case I added to test the `campaign` assignment behaviour. Campaigns are configured in the RRCP with a common prefix. We can then set up a corresponding test on the support site by creating a test named with that prefix.